flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1261,6 +1269,10 @@ private void completePendingCheckpoint(PendingCheckpoint pendingCheckpoint)
                     completedCheckpoint.getTimestamp(),
                     extractIdIfDiscardedOnSubsumed(lastSubsumed));
         }
+
+        // reset the force full snapshot flag, we should've completed at least on full snapshot by
+        // now
+        this.forceFullSnapshot = false;
```

Comment:
If a savepoint is completed before triggering any checkpoint, this flag would be reset as `false`, which is not as expected.

Reply:
That is a very good point! I fixed it and added a test for it.